packages/lib/src/vis/shared/SelectionMesh.tsx

```diff
@@ -63,24 +61,22 @@ function SelectionMesh(props: Props) {
       }
       const { sourceEvent, unprojectedPoint } = evt;
       const { target, pointerId } = sourceEvent;
-      const point = worldToData(unprojectedPoint);
+      const point = new Vector2(unprojectedPoint.x, unprojectedPoint.y);
 
       (target as Element).releasePointerCapture(pointerId);
-      setEndPoint(point);
       setDrag(false);
       if (onSelection && startPoint) {
-        onSelection(startPoint, point);
+        onSelection({ startPoint, endPoint: point });
+        setEndPoint(undefined);
       }
     },
-    [onSelection, startPoint, worldToData, isDragging]
+    [onSelection, startPoint, isDragging]
   );
 
   return (
     <VisMesh {...{ onPointerMove, onPointerUp, onPointerDown, ...meshProps }}>
       <meshBasicMaterial opacity={0} transparent />
-      {startPoint &&
-        endPoint &&
-        children(dataToWorld(startPoint), dataToWorld(endPoint))}
+      {startPoint && endPoint ? children({ startPoint, endPoint }) : null}
```

Comment:
Unsure if setting `null` if the points are undefined should be the responsibility of `SelectionMesh` or its consumers

Reply:
Oh it's actually better like this, I reckon, you're right. Less to think about in the consumer.